src/infrastructure/marketplace.repository.ts

```diff
@@ -37,7 +37,25 @@ export default class MarketplaceRepository implements MarketplaceRepositoryInter
    * @param tool - tool data
    */
   public async addTool(tool: NewToolData): Promise<EditorTool> {
-    const res = await this.transport.post<{ data: EditorTool }>('/editor-tools/add-tool', tool);
+    const data = new FormData();
```

Comment:
it is a transport-level staff, repo should not care about it. Move it to the transport